### ISSUE TITLE
Router doesn't accept valid input

### ISSUE THREAD

**Issue by Mithras** [apollographql/router#1367]
**Describe the bug**
Starting 0.10.0 for schema
```graphql
input MyInput
{
  myField: [Int!] # Note, myField is a nullable array
}
type Query
{
  something (input: MyInput!): String
}
```
Router rejects
```json
{
  "operationName": "MyQuery",
  "variables": {
    "input": {}
  },
  "query": "query MyQuery($input: MyInput!) { something(input: $input) } }"
}
```
also rejects
```json
{
  "operationName": "MyQuery",
  "variables": {
    "input": {
      "myField": null
    }
  },
  "query": "query MyQuery($input: MyInput!) { something(input: $input) } }"
}
```
but accepts
```json
{
  "operationName": "MyQuery",
  "variables": {
    "input": {
      "myField": []
    }
  },
  "query": "query MyQuery($input: MyInput!) { something(input: $input) } }"
}
```

**Expected behavior**
Inputs with nullable array fields (`input SomeInput { someField: [SomeType!] }`) should not fail validation when the field is omitted (`{ }`) or null (`{ "someField": null }`).

**Output**
```json
{
  "errors": [
    {
      "message": "invalid type for variable: 'input'",
      "locations": [],
      "path": null,
      "extensions": {
        "type": "ValidationInvalidTypeVariable",
        "name": "input"
      }
    }
  ]
}
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Version: 0.10.0

**Additional context**
Probably related to https://github.com/apollographql/router/pull/1211
Also, the error message is pretty much useless and I had to figure out what the Router doesn't like by pure elimination.

**Comment by Mithras** [comment]
And a suggestion, please add unit tests for all issues you are fixing. Last 3 versions I tried to upgrade to were all broken.

**Comment by Geal** [comment]
I think this was fixed in https://github.com/apollographql/router/pull/1363, but that was done right after the 0.10 release

>  And a suggestion, please add unit tests for all issues you are fixing. Last 3 versions I tried to upgrade to were all broken.
We do, especially for issues related to GraphQL validation. Which specific issues are you referring to?

**Comment by Mithras** [comment]
>We do, especially for issues related to GraphQL validation. Which specific issues are you referring to?

https://github.com/apollographql/router/issues/928 was fixed but then https://github.com/apollographql/router/issues/1286 happened. Now https://github.com/apollographql/router/issues/1286 is fixed but this issue happened. So I'm unable to find a working version for a while.

**Comment by Mithras** [comment]
Well, I build `866b84807f14dcd9ebe9dfb1ef76b411931d6110` myself and it seems to be working so far.

**Comment by Geal** [comment]
> @Geal what's the official release cadence?

We try to get a release per week, but it varies if we are waiting on some PRs, or just if people are plainly busy. We have not committed to a strict release schedule for now.

> https://github.com/apollographql/router/issues/928 was fixed but then https://github.com/apollographql/router/issues/1286 happened. Now https://github.com/apollographql/router/issues/1286 is fixed but this issue happened. So I'm unable to find a working release for a while.

Right, #928 was incomplete so it needed further fixes in #1286 and #1360. Fragment application with subtyping is a tricky part of the specification. Input validation was mostly done (there's a huge number of tests for it), but that one escaped our attention.

It feels a bit too reactive to my tests, but I can assure you we have spent significant time on specification edge cases, built [tooling to check for them](https://github.com/apollographql/router/tree/main/fuzz), and take them seriously (most of those bugs are fixed in a day or two then released the next week).
Anyway I'm planning to go through the specification again soon and see if we could have missed something else.

> Well, I built 866b84807f14dcd9ebe9dfb1ef76b411931d6110 myself and it seems to be working so far.

Great! I'm closing this then, don't hesitate to come back for more questions or issues :)
